File: src/world.js
const redis = require('./redis')

module.exports = async ({world}, req) => ({
    id: world,
    name: async () => await redis.hget(`world:${world}`, 'name'),
    towns: async () => (await redis.hvals(`world:${world}:towns`))
        .map((s) => require('./town')({town: s}))
})

File: lib/towns_kings/faktory/world.ex
require TownsKings.Repo.Macro.WorkList
import TownsKings.Repo.Macro.WorkList

define_object TownsKings.Repo.World do

  redis :world,
        [
          {:self, [:name]},
          {:towns, :hset}
        ]

  g_prov worlds() do
    {:ok, r} = Redix.command(:redix, ["smembers", "worlds"])

    r
  end

  prov towns() do
    {:ok, r} = Redix.command(:redix, ["hvals", "world:#{@self}:towns"])

    r
  end

end

